tests/test_py37.py

```diff
@@ -246,3 +246,89 @@ class Node(BaseModel):
     node = Node(**data)
     assert isinstance(node.left, Leaf)
     assert isinstance(node.right[0], Node)
+
+
+@skip_not_37
+def test_self_reference_json_schema(create_module):
+    module = create_module(
+        """
+from __future__ import annotations
+from typing import List
+from pydantic import BaseModel, Schema
+
+class Account(BaseModel):
+  name: str
+  subaccounts: List[Account] = []
+
+Account.update_forward_refs()
+    """
+    )
+    Account = module.Account
+    assert Account.schema() == {
+        "$ref": "#/definitions/Account",
```

Comment:
single quotes please.

Reply:
Yep. Sorry for that. I forget to use them and forget the format doesn't change them.

I also fixed some double quotes above I think I had left in a previous PR.

Reply:
great thanks, it's annoy there's no way to fix or even lint for this.

Even if you patch black to prefer single quotes to double, it messes up too many strings to use.

Reply:
Yeah, maybe we can add something else to check for single quotes.

And show a linting error or something (even if not autoformatting it). Maybe something like https://github.com/zheller/flake8-quotes.

But that's another PR...